webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/facts/nodes/NodeFactRepository.scala

```diff
@@ -576,30 +603,80 @@ class CoreNodeFactRepository(
     }
   }
 
-  def save(
+  /*
+   * Save can be used in two fashion: either for setting tenants
+   * or for save. In the second case, we must ensure tenants don't change.
+   * In the first case, we use the existing coreNodeFact for save.
+   */
+  private def internalSave(
```

Comment:
we factor save infrastructure logic (atomicity, call to backend storage, change event etc) to both save and setSecurityTag. 
The node parameter will chose in what mode we operate.